src/streamlink/plugins/rotana.py

```diff
@@ -5,35 +5,97 @@
 $region Saudi Arabia
 """
 
+import base64
 import logging
 import re
 
 from streamlink.plugin import Plugin, pluginmatcher
+from streamlink.plugin.api import validate
 from streamlink.stream.hls import HLSStream
 
 log = logging.getLogger(__name__)
 
 
 @pluginmatcher(re.compile(
-    r'https?://(?:www\.)?rotana\.net/live'
+    r"https?://(?:www\.)?rotana\.net/"
 ))
 class Rotana(Plugin):
-    _hls_re = re.compile(r'''["'](?P<url>[^"']+\.m3u8)["']''')
+    js_url_re = re.compile(r"""['"](https://hiplayer.hibridcdn.net/l/[^'"]+)['"]""")
+    base64_data_re = re.compile(r"i\s*=\s*\[(.*)\]\.join")
 
     def _get_streams(self):
-        res = self.session.http.get(self.url)
-        m = self._hls_re.search(res.text)
-        if not m:
-            log.debug('No video URL found.')
+        js_url = self.session.http.get(
+            self.url,
+            schema=validate.Schema(
+                validate.parse_html(),
+                validate.xml_xpath_string(".//script[contains(text(), 'https://hiplayer.hibridcdn.net/l/')]/text()"),
+                validate.any(
+                    None,
+                    validate.all(
+                        validate.transform(self.js_url_re.search),
+                        validate.any(
+                            None,
+                            validate.all(
+                                validate.get(1),
+                                validate.url(),
+                            ),
+                        ),
+                    ),
+                ),
+            ),
+        )
+
+        if not js_url:
             return
 
-        hls_url = m.group('url')
-        log.debug('URL={0}'.format(hls_url))
-        streams = HLSStream.parse_variant_playlist(self.session, hls_url)
-        if not streams:
-            return {'live': HLSStream(self.session, hls_url)}
-        else:
-            return streams
+        log.debug(f"JS URL={js_url}")
+
+        data = self.session.http.get(
+            js_url,
+            schema=validate.Schema(
+                validate.transform(self.base64_data_re.search),
+                validate.any(
+                    None,
+                    validate.all(
+                        validate.get(1),
+                        validate.transform(lambda s: s.replace("'", "")),
+                        validate.transform(lambda s: s.replace('"', "")),
+                        validate.transform(lambda s: s.replace(",", "")),
+                        validate.transform(lambda s: s.replace(" ", "")),
```

Comment:
```suggestion

                        validate.transform(lambda s: re.sub(r"['\", ]", "", s)),
```

Or replace the negation of the specific base64 alphabet...